test/c/test_substrait_c_api.cpp

```diff
@@ -320,3 +320,281 @@ TEST_CASE("Test C VirtualTable input Expression", "[substrait-api]") {
   REQUIRE(CHECK_COLUMN(result, 0, {2, 6}));
   REQUIRE(CHECK_COLUMN(result, 1, {4, 8}));
 }
+
+TEST_CASE("Test C Iceberg Substrait with Substrait API", "[substrait-api][iceberg][interesting]") {
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	CreateEmployeeTable(con);
+
+	REQUIRE_NO_FAIL(con.Query("INSTALL iceberg;"));
+	REQUIRE_NO_FAIL(con.Query("LOAD iceberg;"));
+
+        const string plan_json = R"plan(
+		{
+		  "extensionUris" : [ {
+		    "extensionUriAnchor" : 1,
+		    "uri" : "https://github.com/substrait-io/substrait/blob/main/extensions/functions_aggregate_generic.yaml"
+		  } ],
+		  "extensions" : [ {
+		    "extensionFunction" : {
+		      "extensionUriReference" : 1,
+		      "functionAnchor" : 1,
+		      "name" : "count"
+		    }
+		  } ],
+		  "relations" : [ {
+		    "root" : {
+		      "input" : {
+		        "project" : {
+		          "input" : {
+		            "aggregate" : {
+		              "input" : {
+		                "read" : {
+		                  "baseSchema" : {
+		                    "names" : [ "fruit", "count" ],
+		                    "struct" : {
+		                      "types" : [ {
+		                        "string" : {
+		                          "nullability" : "NULLABILITY_NULLABLE"
+		                        }
+		                      }, {
+		                        "decimal" : {
+		                          "scale" : 0,
+		                          "precision" : 10,
+		                          "nullability" : "NULLABILITY_NULLABLE"
+		                        }
+		                      } ],
+		                      "nullability" : "NULLABILITY_REQUIRED"
+		                    }
+		                  },
+		                  "icebergTable" : {
+		                    "direct" : {
+		                      "metadataUri" : "data/iceberg",
```

Comment:
i don't think that is a manifest file location, i think it is a directory. the substrait api doesn't support using directories at this point so we shouldn't support it implicitly here.

Reply:
Updated test.  Required creating an iceberg table that is relative instead of absolute.